app/pages/nonsipp/memberdetails/MemberDetailsPage.scala

```diff
@@ -36,16 +43,65 @@ case class MemberDetailsPage(srn: Srn, index: Max300) extends QuestionPage[NameD
     (value, userAnswers.get(this)) match {
       case (None, _) =>
         //deletion
-        removePages(userAnswers, pages(srn))
+        userAnswers.remove(pages(srn, userAnswers) ::: paymentsPages(srn, index, userAnswers))
       case _ => Try(userAnswers)
     }
 
-  private def pages(srn: Srn): List[Removable[_]] =
+  private def pages(srn: Srn, userAnswers: UserAnswers): List[Removable[_]] =
     List(
       DoesMemberHaveNinoPage(srn, index),
       MemberStatus(srn, index)
     )
 
+  private def paymentsPages(srn: Srn, index: Max300, userAnswers: UserAnswers): List[Removable[_]] = {
```

Comment:
What should happen if the last member is removed? Should member payments journey type pages get removed (e.g.  TransfersInJourneyStatus) so that the tasklist and the data reverts back to the state how it was before adding any members?

Reply:
I've added a solution to this now - all member related pages that are remaining are removed upon the last member being deleted.